packages/chatbot-server-mongodb-public/src/tracing/routesUpdateTraceHandlers.ts

```diff
@@ -32,6 +57,36 @@ export const makeAddMessageToConversationUpdateTrace: (
       tracingData.tags.push(`auth_user`);
     }
 
+    const userMessage = tracingData.userMessage;
+    const { userId, anonymousId } = getSegmentIds(userMessage);
+    if (userMessage) {
+      segmentTrackUserSentMessage?.({
+        userId,
+        anonymousId,
+        conversationId: conversation._id,
+        origin: userMessage.customData?.origin as string,
+        createdAt: userMessage.createdAt,
+        tags: tracingData.tags,
+      });
+    }
+
+    const assistantMessage = tracingData.assistantMessage;
+    if (userMessage && assistantMessage) {
+      segmentTrackAssistantResponded?.({
+        userId,
+        anonymousId,
+        conversationId: conversation._id,
+        origin: userMessage.customData?.origin as string,
+        createdAt: assistantMessage.createdAt,
+        isVerifiedAnswer: tracingData.isVerifiedAnswer ?? false,
+        rejectionReason: tracingData.rejectQuery
+          ? (assistantMessage.customData?.rejectionReason as
+              | string
+              | undefined) ?? "Unknown rejection reason"
+          : undefined,
+      });
+    }
+
     logger.updateSpan({
```

Comment:
now that there are 2 separate things happening in this trace func, should we wrap each w/ a try/catch block? 

i know both aren't async, since the processing happening in background..but still. 

not sure what right approach is, but wanted to raise the question